dtrace_provider.cc

```diff
@@ -92,6 +92,10 @@ namespace node {
     DTraceProbe *probe = Nan::ObjectWrap::Unwrap<DTraceProbe>(pd->ToObject());
     obj->Set(info[0]->ToString(), pd);
 
+    // reference the provider to avoid GC'ing it when only probes remain in scope.
+    Nan::ForceSet(pd, Nan::New<String>("__prov__").ToLocalChecked(), obj,
```

Comment:
Is this a standard way to add a reference?  I'm not sure I follow exactly what's going on here.